agent.go

```diff
@@ -278,6 +279,8 @@ func (agent *agentS) fullRequestResponse(url string, method string, data interfa
 						ret = resp.Header.Get(header)
 					}
 				}
+
+				io.Copy(ioutil.Discard, resp.Body)
```

Comment:
Makes sense, let's put a limit here though, to avoid reading unreasonably large responses though:
```suggestion
				io.CopyN(ioutil.Discard, resp.Body, 256 << 10)
```
The constant comes from [here](https://golang.org/src/net/http/server.go#L1097) and corresponds an average TCP buffer size.

Reply:
Good point!